src/Post/CommentListItem.jsx

```diff
@@ -0,0 +1,69 @@
+import { useContext, useEffect, useState } from "react"
+import { deleteComment, getAuthor, updateComment } from "../Api"
+import { useNavigate } from "react-router-dom"
+import { CommentContext } from "../PostListItem"
+
+let InitialAuthor = {
+    firstName: "",
+    lastName: "",
+    gender: "",
+    email: "",
+    jobTitle: "",
+}
+
+function CommentListItem({ comment }) {
+    const [author, setAuthor] = useState({ ...InitialAuthor })
+    const [isEditing, setIsEditing] = useState(false)
+    const { comments, setComments } = useContext(CommentContext)
+    const navigate = useNavigate()
+    useEffect(() => {
+        getAuthor(comment.contactId)
+            .then((response) => { setAuthor(response) })
+    }, [comment])
+
+    const removeComment = (event) => {
+        deleteComment(comment.postId, comment.id)
+        comments.splice(comments.indexOf(comment), 1)
```

Comment:
Directly mutating the comments array with splice and then updating state with setComments([...comments]) is not recommended. Instead, use functional updates or methods that return a new array to ensure immutability